src/components/UserProfile/UserProfile.jsx

```diff
@@ -0,0 +1,94 @@
+import axios from "axios";
+import { useEffect, useState } from "react"
+import { useParams } from "react-router-dom";
+import { BACKEND_URL } from "../../constants/constants";
+import Loader from "../Loader/Loader";
+import NotFound from "../NotFound/NotFound";
+import PartyCard from "../PartyCard/PartyCard";
+import "./UserProfile.css"
+
+export default function UserProfile() {
+  const params = useParams();
+
+  const [parties, setParties] = useState(null);
+  const [user, setUser] = useState(undefined)
+  const [error, setError] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  useEffect( () => {
+    const getProfile = async () => {
+      try{
+        const res = await axios.get(`${BACKEND_URL}user/${params.userId}/profile`)
+        setUser(res.data.data.user)
+        setParties(res.data.data.parties)
+        setLoading(false)
+      }
+      catch (err) {
+        console.error(err)
+        setError(err);
+        if(err.response.status===404) {
+          setUser("Not Found")
+        }
+        else {
+          setUser(null);
+          setError(err)
+        }
+        setLoading(false)
+      }
+    }
+    setLoading(true)
+    getProfile()
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+  if(loading) {
+    return (
+      <Loader />
+    )
+  }
+  if(user==="Not Found") {
+    return(<NotFound />)
+  }
+
+  if(error) {
+    return (
+      <div className="user-failed"> 
+        <h1 className="user-failed-message">{error.response.data ? error.response.data.error.message : error.message}</h1>
+        <h2 className="user-failed-status-text">{error.statusText}</h2>
+      </div>
+    )
+  }
+
+  if(user===undefined) {
```

Comment:
if you want to guard against `null` also, you can use == as in `user == null`.  Not necessary though.